pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kubeflowjob
+
+import (
+	"strings"
+
+	kftraining "github.com/kubeflow/training-operator/pkg/apis/kubeflow.org/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/maps"
+)
+
+type KubeflowJob struct {
+	KFJobControl KFJobControl
+}
+
+var _ jobframework.GenericJob = (*KubeflowJob)(nil)
+var _ jobframework.JobWithPriorityClass = (*KubeflowJob)(nil)
+
+func (j *KubeflowJob) Object() client.Object {
+	return j.KFJobControl.Object()
+}
+
+func (j *KubeflowJob) IsSuspended() bool {
+	return j.KFJobControl.RunPolicy().Suspend != nil && *j.KFJobControl.RunPolicy().Suspend
+}
+
+func (j *KubeflowJob) Suspend() {
```

Comment:
Can RunPolicy be nil? Should we add it here if it's missing?

Reply:
RunPolicy isn't a pointer. So we wouldn't be concerned about missing Runpolicy.

https://github.com/kubeflow/training-operator/blob/48dbbf0a8e90e52c55ec05d0f689fcbf83c6b441/pkg/apis/kubeflow.org/v1/pytorch_types.go#L76-L80

@alculquicondor WDYT?

Reply:
@alculquicondor friendly ping :)

Reply:
sounds good :)